src/flint/fmpz.jl

```diff
@@ -1835,6 +1839,83 @@ function rand_bits_prime(::FlintIntegerRing, n::Int, proved::Bool = true)
    return z
 end
 
+# rand in a range
+# this mirrors the implementation for BigInt in the Random module
+
+function as_bigint(z::fmpz)
+   @assert coeff_is_mpz(z)
+   unsafe_load(Ptr{BigInt}(z.d << 2))
```

Comment:
This is unsafe unless the caller does a `GC.@preserve z` right? I see that you did this in one of the functions, but not in the other.

Anyway, this should be hidden by more underscores.

Reply:
Yes right, and I will add underscore.